Not in scope: type constructor or class ‘Prim.Any’ (GHC HEAD)

Compiling constraints with GHC HEAD, I get an error about `Prim.Any`:

```bash
bash-3.2$ cabal install constraints
Resolving dependencies...
Configuring constraints-0.8...
Building constraints-0.8...
Failed to install constraints-0.8
Build log ( /Users/conal/.cabal/logs/constraints-0.8.log ):
cabal: Entering directory '/var/folders/wx/m2wn7shj0gn4q_y_wnybcsqh0000gn/T/cabal-tmp-76236/constraints-0.8'
Configuring constraints-0.8...
Building constraints-0.8...
Preprocessing library constraints-0.8...
[1 of 5] Compiling Data.Constraint  ( src/Data/Constraint.hs, dist/build/Data/Constraint.o )

src/Data/Constraint.hs:308:7: error:
    Not in scope: type constructor or class ‘Prim.Any’
    Module ‘GHC.Prim’ does not export ‘Any’.
cabal: Leaving directory '/var/folders/wx/m2wn7shj0gn4q_y_wnybcsqh0000gn/T/cabal-tmp-76236/constraints-0.8'
cabal: Error: some packages failed to install:
constraints-0.8 failed during the building phase. The exception was:
ExitFailure 1
bash-3.2$ ghc --version
The Glorious Glasgow Haskell Compilation System, version 8.1.20160405
```

constraints-0.8 seems to have compiled fine with GHC 8.1.20160307.

--- a/src/Data/Constraint.hs
+++ b/src/Data/Constraint.hs
@@ -91,7 +91,7 @@ import GHC.Prim (Constraint)
 #else
 import GHC.Types (Constraint)
 #endif
-import qualified GHC.Base (Any)
+import qualified GHC.Exts (Any)
 
 
 -- | Values of type @'Dict' p@ capture a dictionary for a constraint of type @p@.
@@ -306,7 +306,7 @@ top :: a :- ()
 top = Sub Dict
 
 -- | 'Any' inhabits every kind, including 'Constraint' but is uninhabited, making it impossible to define an instance.
-class GHC.Base.Any => Bottom where
+class GHC.Exts.Any => Bottom where
   no :: Dict a
 
 -- |

--- a/src/Data/Constraint.hs
+++ b/src/Data/Constraint.hs
@@ -91,7 +91,8 @@ import GHC.Prim (Constraint)
 #else
 import GHC.Types (Constraint)
 #endif
-import qualified GHC.Prim as Prim
+import qualified GHC.Base (Any)
+
 
 -- | Values of type @'Dict' p@ capture a dictionary for a constraint of type @p@.
 --
@@ -305,7 +306,7 @@ top :: a :- ()
 top = Sub Dict
 
 -- | 'Any' inhabits every kind, including 'Constraint' but is uninhabited, making it impossible to define an instance.
-class Prim.Any => Bottom where
+class GHC.Base.Any => Bottom where
   no :: Dict a
 
 -- |

